### ISSUE TITLE
Example for accessing and mutating server state in RPC calls?

### ISSUE THREAD

**Issue by mystal** [paritytech/jsonrpc#224]
All the examples I've found so far don't have any real state on the server that is accessed or mutated in RPC calls. I went digging and found the emerald project which seems to do some of that here:
https://github.com/ethereumproject/emerald-rs/blob/master/src/rpc

That seems like a reasonable approach to take, using Arc<Mutex<ServerState>> to access and mutate server state. Does that sound right?

**Comment by tomusdrw** [comment]
Yes, the state needs to be wrapped in a `Mutex` (`Arc` is not necessary afair, unless you want to have access to that state outside of the RPC server implementation).

**Comment by tomusdrw** [comment]
Actually it's a fair point that we should add some more real-life examples.

**Comment by shesek** [comment]
I've been wondering about this myself as well. Doing this with `io.add_method` and closure state seems easy enough, but how should one go about handling state in `jsonrpc-derive` methods?
